hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -3061,3 +3075,26 @@ func reconcileMachineApproverDeployment(deployment *appsv1.Deployment, hc *hyper
 	hyperutil.SetDefaultPriorityClass(deployment)
 	return nil
 }
+
+// isGroupVersionRegistered determines if a specified groupversion is registered on the cluster
+func isGroupVersionRegistered(client *discovery.DiscoveryClient, groupVersion schema.GroupVersion) (bool, error) {
+	_, apis, discoveryError := client.ServerGroupsAndResources()
+	if discoveryError != nil && !discovery.IsGroupDiscoveryFailedError(discoveryError) {
```

Comment:
How about:

```
	_, apis, err := client.ServerGroupsAndResources()
	if err != nil {
		if discovery.IsGroupDiscoveryFailedError(err) {
			// If the group we are looking for can't be fully discovered,
			// that does still mean that it exists
			e := err.(*discovery.ErrGroupDiscoveryFailed)
			if _, exists := e.Groups[groupVersion]; exists {
				return true, nil
			}
		}
		return false, err
	}
```

Reply:
I think it would have to be 


```go
	_, apis, err := client.ServerGroupsAndResources()
	if err != nil {
		if discovery.IsGroupDiscoveryFailedError(err) {
			// If the group we are looking for can't be fully discovered,
			// that does still mean that it exists
			e := err.(*discovery.ErrGroupDiscoveryFailed)
			if _, exists := e.Groups[groupVersion]; exists {
				return true, nil
			}
		} else {
		    return false, err
		}
	}
```

because we want to continue the normal discovery if we get a `ErrGroupDiscoveryFailed` but yes I can change that. :+1
Will cover relevant cases in the unit tests.

Reply:
So if we get a `ErrGroupDiscoveryFailed` error and
```
		if _, exists := e.Groups[groupVersion]; exists {
			return true, nil
		}
```
does not return true, we still need to run
```
	for _, api := range apis {
		if api.GroupVersion == groupVersion.String() {
			return true, nil
		}
	}
```
?